airscan-wsd.c

```diff
@@ -87,6 +87,13 @@ typedef struct {
     bool          png;
     bool          dib;
 
+    /* Supported content types */
+    bool          content_auto;
```

Comment:
Why use these bools?

I use it for WSD formats, because WSD supports 2 variants of JPEG and 4 variants of tiff, which I report upstack as a single JPEG/TIFF bit, and map back to the suitable variant in the opposite direction.

But here we have 1:1 mapping, so we can simple decode supported content types directly into the bitmap (like colormodes, for example)

Reply:
That's what I get for blindly copying for a quick initial hack :)